pkg/api/api.go

```diff
@@ -67,6 +67,148 @@ import (
 // loggerName is the tree path name of the logger for this package.
 const loggerName = "api"
 
+type headerDescriptor struct {
+	// name of header parameter
+	Name string
+
+	// Validate will validate parameter value or
+	// return error if value is not in correct format.
+	Validate func(v interface{}, s *Service) error
```

Comment:
Why is the `v` interface{} type when headers are always strings?

Reply:
sorry, i was playing around and focused on other part and didn't give it much thought before. Will be fixed in next commit.